src/shared/ui/Badge/Badge.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+type Props = {
+  /**
+   * 제출 여부를 나타냅니다.
+   * - `true`: 제출 완료 (green)
+   * - `false`: 미제출 (red)
+   * - `null`: 제출 전 (gray)
+   */
+  isSubmited?: boolean | null;
+
+  /**
+   * 진행 상태를 나타냅니다.
+   * - `true`: 진행 중 (green)
+   * - `false`: 마감 (red)
+   * - `null`: 진행 전 (gray)
+   */
+  isProgress?: boolean | null;
+
+  /**
+   * 모집 상태를 나타냅니다.
+   * - `true`: 모집 중 (green)
+   * - `false`: 모집 마감 (gray)
+   */
+  isRecruiting?: boolean;
+
+  /**
+   * - `true`: 초록색 (`bg-green-100 text-green-300`)
+   * - `false`: 빨간색 (`bg-red-100 text-red-300`)
+   * - `null`: 회색 (`bg-gray-100 text-gray-400`)
+   *
+   * @default null
+   */
+  color?: boolean | null;
+
+  /**
+   * 특정 상태가 없을 경우, text값이 표시됩니다.
+   *
+   * @default '상태 없음'
+   */
+  text?: string;
+};
+
+/**
+ * `Badge` is a UI element that visually represents various statuses.
+ */
+export function Badge({ isSubmited, isProgress, isRecruiting, color, text }: Props) {
```

Comment:
```suggestion
} & React.HTMLAttributes<HTMLDivElement>;

/**
 * `Badge` is a UI element that visually represents various statuses.
 */
export function Badge({ isSubmited, isProgress, isRecruiting, color, text, ...props  }: Props) {
```
div 속성을 상속받아 더 유연하게 작업할 수 있을 것 같아요 !